### PR TITLE
[e2e] Cypress scripts improvements

### PR DESCRIPTION
**Pull request by tuliobluz**
PR Checklist

- [X] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Developer experience (improves developer workflows for contributing to the project)

## Description

This PR is to improve the Cypress scripts

- Make sure the queryDoc gets the right test doc using the generated username` triggeredBy.userId === visitor.username,.` for the discussions scenarios.
- Add to how to write scenarios dynamic generate title then we can query the right document based on the dynamic title
- Use `should(be.visible)` instead `should(exist)`
- Removed duplicate `cy.clearServiceWorkers()` that was in `after()` and also in `before()` on `hooks.ts`
- Remove `cy.wait(500)` and used `.focus()` to type the intro description on `research/write.ts`

### CODE REVIEW

**Review by mariojsnunes — APPROVED** [review]

**Comment by benfurber** [comment]
@tuliobluz Looks like the current cypress fails might be related to your changes? Will leave you to confirm/work on before reviewing again.

**Comment by tuliobluz** [comment]
Hello @benfurber I just pushed new commits that might fix all the scenarios. Can you please retry the unit tests? Hopefully, if everything passes in the CI you can review.

File: packages/cypress/src/integration/research/write.spec.ts [hunk 1]
```diff
@@ -80,8 +80,13 @@ describe('[Research]', () => {
...
 
-      cy.wait(500) // wierd issue where description is cleared during typing below (causes first few characters to not be typed, thus failing the test)
-      cy.get('[data-cy=intro-description]').type(expected.description).blur()
+      // workround to type full text as it is an issue https://github.com/cypress-io/cypress/issues/3817
+      cy.get('[data-cy=intro-description]')
+        .wait(0)
+        .focus()
+        .clear()
+        .type(expected.description)
+        .blur()
```
[the remark this answers is not in the document]
Comment (tuliobluz) [reply packages/cypress/src/integration/research/write.spec.ts]: I ran it several times locally and it worked, as the ci is still facing the problem, I guess we have to use the `cy.wait`

**Comment by tuliobluz** [comment]
Please don't merge, I am still investigating the flaky regarding the delete step on `howto/write.spec`

**Comment by benfurber** [comment]
@tuliobluz Are you sure this can't be merged in @tuliobluz, `howto/write` is already a little annoying.

**Comment by tuliobluz** [comment]
@benfurber Lets merge, and I will continue working on it

**Pull request merged**